src/app/pages/4_Find_Assay_Target_Area.py

```diff
@@ -0,0 +1,134 @@
+import streamlit as st
+
+from app.common.constants import TGT_AREA_DF, TGT_AREA_FORM
+from app.common.setup import init_session_state_tgt_area
+from genetic_testing.assay_target.datatypes import AssayTargetColumns
+from genetic_testing.assay_target.primer_design import find_target_area
+
+# Initialize the Streamlit session state for this page
+init_session_state_tgt_area()
+
+# Define constants for download options
+CSV_DOWNLOAD = "Download as CSV"
+FASTA_DOWNLOAD = "Download sequences as a fasta file"
+
+# Streamlit app header
+st.header("Find Assay Target Area")
+
+# Create a container for uploading target sequences
+target_container = st.container()
+with target_container:
+    st.subheader("Upload Target Sequences")
+    # Create a place to upload FASTA files
+    target_files = st.file_uploader(
+        "Upload FASTA files",
+        key="target_files_uploader",
+        type=["fasta", "fas"],
+        accept_multiple_files=True,
+        help="Only single FASTA or non-aligned multi-FASTA files are allowed",
+    )
+
+st.divider()
+
+# Create a container for uploading off-target sequences
+off_target_container = st.container()
+with off_target_container:
+    st.subheader("Upload Off-Target Sequences")
+
+    # Create a place to upload FASTA files
+    off_target_files = st.file_uploader(
+        "Upload FASTA files",
+        key="off_target_files_uploader",
+        type=["fasta", "fas"],
+        accept_multiple_files=True,
+        help="Only single FASTA or non-aligned multi-FASTA files are allowed",
+    )
+
+st.divider()
+
+# Create a container for target area parameters
+target_area_container = st.container()
+with target_area_container:
+    st.subheader("Target Area Parameters")
+    with st.form("find_target_area"):
+        # Input field for selecting Reference sequence, Target region Size, Target region Slide size, and Maximum allowed differences between primers and targets
+        reference_sequence = st.text_input(
+            "Select Reference Sequence",
+            # [file.name for file in target_files],
+            help="If no selection is made, the first file will be used as the reference sequence",
```

Comment:
Would it make sense to pre-populate this field with the first header/file? That way it's clear what will be used without reading the help but can still be changed by the user. Also, should this be a drop-down instead of open text? What happens if there's a typo in the name of the reference sequence?

Reply:
Yes, I had tried this earlier to pre-populate this field with the first header/file and it worked well. But, in order to do that I will have to parse the files and read the headers from each file. Since, Streamlit runs the page on every interaction, we will have to perform this step again and again. We can make use of `st.cache_data` [here](https://docs.streamlit.io/library/advanced-features/caching#:~:text=from%20st.cache-,Minimal%20example,-To%20cache%20a) and ensure that we cache these header names.

I thought about using `st.cache_data` but I wanted to get this feature out before Misa left, so I went ahead with a simpler approach.

After we resolve all the issues, we can take this task up? What do you think?

Reply:
Gotcha, yeah, makes sense. Can you log an issue as a feature suggestion so Misa can weigh on it?

Reply:
Okay, I have created an issue and assigned that issue to you - https://github.com/uw-ssec/neglected-diagnostics/issues/55